### PR TITLE
MODELIX-721 Migrate cucumber tests for model-server to ktor tests

### PR DESCRIPTION
**Pull request by mhuster23**
On my local machine the migrated test cases now only take 2s instead of 38s.
The saved 36s are roughly a 40% reduction of the previous total time to test the model-server.

## To be verified by reviewers

* [ ] Relevant public API members have been documented
* [ ] Documentation related to this PR is complete
  * [ ] Boundary conditions are documented
  * [ ] Exceptions are documented
  * [ ] Nullability is documented if used
* [ ] Touched existing code has been extended with documentation if missing
* [ ] Code is readable
* [ ] New features and fixed bugs are covered by tests

### CODE REVIEW

File: model-server/src/main/kotlin/org/modelix/model/server/handlers/ui/IndexPage.kt [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+import io.ktor.server.routing.routing
+import kotlinx.html.a
+import kotlinx.html.h1
+import kotlinx.html.li
+import kotlinx.html.style
+import kotlinx.html.ul
+import kotlinx.html.unsafe
+import org.modelix.model.server.templates.PageWithMenuBar
+
+class IndexPage {
```
[the remark this answers is not in the document]
Comment (languitar) [reply model-server/src/main/kotlin/org/modelix/model/server/handlers/ui/IndexPage.kt:33]: Let's just add a few word here.

File: model-server/src/test/kotlin/org/modelix/model/server/V1ApiTest.kt [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+        assertEquals(HttpStatusCode.OK, response2.status)
+        assertNotEquals(response1.bodyAsText(), response2.bodyAsText())
+    }
+
+    @Test
+    fun `events are received after subscription`() = runTest {
+        val key = "dylandog"
+        val value = "a comic book"
+
+        coroutineScope {
```
Comment (languitar) [inline model-server/src/test/kotlin/org/modelix/model/server/V1ApiTest.kt:75]: Why do these tests need an explicit scope?

File: model-server/src/test/kotlin/org/modelix/model/server/handlers/HealthApiTest.kt [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+        }
+
+        block()
+    }
+
+    @Test
+    fun `health endpoint returns healthy`() = runTest {
+        val response = client.get("/health")
+        assertEquals("healthy", response.bodyAsText())
+    }
```
Comment (languitar) [inline model-server/src/test/kotlin/org/modelix/model/server/handlers/HealthApiTest.kt:56]: For the sake of completeness, another case could be added non to see if unhealthy is returned when something is off. Also, the return code should be validated here.

File: model-server/src/test/kotlin/org/modelix/model/server/V1ApiTest.kt
Comment (languitar) [inline model-server/src/test/kotlin/org/modelix/model/server/V1ApiTest.kt:1]: Can we use this PR as an opportunity to clean up the test case names for matching the style we use elsewhere. For instance "retrieving multiple keys - all existing" could become "All existing keys can be pulled/retrieved".

**Review by languitar — COMMENTED** [review]
Do we know the reasons for why some of the cucumber scenarios were commented out? Is it safe to drop them?

Apart from that, only minor things that I added.

After #862, kotest assertions would now also be available as an option.

File: model-server/src/main/kotlin/org/modelix/model/server/handlers/ui/IndexPage.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (mhuster23) [reply model-server/src/main/kotlin/org/modelix/model/server/handlers/ui/IndexPage.kt:33]: What kind of information are you missing here? Explaining what an index page is in general feels overkill here.

File: model-server/src/test/kotlin/org/modelix/model/server/V1ApiTest.kt (hunk 2 above)
Comment (mhuster23) [reply model-server/src/test/kotlin/org/modelix/model/server/V1ApiTest.kt:75]: To be able to call `async`. The get request will only be answered once the put has been performed.

File: model-server/src/main/kotlin/org/modelix/model/server/handlers/ui/IndexPage.kt [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+                            li {
+                                a("user") { +"View JWT token and permissions" }
+                            }
+                            li {
+                                a("swagger") { +"SwaggerUI" }
+                            }
+                        }
+                    }
+                }
+                call.respondText("Model Server")
```
Comment (languitar) [inline model-server/src/main/kotlin/org/modelix/model/server/handlers/ui/IndexPage.kt:79]: Btw, what is this line actually doing?

File: model-server/src/test/kotlin/org/modelix/model/server/V1ApiTest.kt (hunk 2 above)
Comment (languitar) [reply model-server/src/test/kotlin/org/modelix/model/server/V1ApiTest.kt:75]: That is because of the execution semantics of `runTest`?

If the intended effect of the whole code here is to achieve:

1. GET starts polling
2. PUT happens
3. GET returns

Then I am currently not sure if that really happens that way.

First, with any general type of dispatcher, the block in the `async` might start executing only after the PUT has happened. So we have no real guarantee that this test is actually testing what it assumes to test.

Second, with the test dispatcher in particular, only the `await` will actually trigger the execution of the `async` block. So we will never end up in GET being started before the PUT. And `coroutineScope` does not magically switch to a different dispatcher as far as I know.

File: model-server/src/main/kotlin/org/modelix/model/server/handlers/ui/IndexPage.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (languitar) [reply model-server/src/main/kotlin/org/modelix/model/server/handlers/ui/IndexPage.kt:33]: Not much ;-) But we'd eventually get to a state with less warnings, which makes it easier to spot the important ones.

File: model-server/src/test/kotlin/org/modelix/model/server/V1ApiTest.kt (hunk 2 above)
Comment (mhuster23) [reply model-server/src/test/kotlin/org/modelix/model/server/V1ApiTest.kt:75]: I have verified that it happens that way. Note that this runs in the default dispatcher since it's not the `runTest` of `kotlinx.coroutines`.
By default async is scheduled for immediate execution. However there might be a race condition, where the PUT might happen before the GET. Though I'm not sure on that, since i haven't dived deep enough into kotlin coroutines yet.

File: model-server/src/main/kotlin/org/modelix/model/server/handlers/ui/IndexPage.kt (hunk 4 above)
Comment (mhuster23) [reply model-server/src/main/kotlin/org/modelix/model/server/handlers/ui/IndexPage.kt]: removed it

File: model-server/src/test/kotlin/org/modelix/model/server/V1ApiTest.kt
Comment (mhuster23) [reply model-server/src/test/kotlin/org/modelix/model/server/V1ApiTest.kt:1]: done

File: model-server/src/main/kotlin/org/modelix/model/server/handlers/ui/IndexPage.kt (hunk 1 above)
[the remark this answers is not in the document]
Comment (mhuster23) [reply model-server/src/main/kotlin/org/modelix/model/server/handlers/ui/IndexPage.kt:35]: done

File: model-server/src/test/kotlin/org/modelix/model/server/V1ApiTest.kt (hunk 2 above)
Comment (languitar) [reply model-server/src/test/kotlin/org/modelix/model/server/V1ApiTest.kt:75]: Can we rename `runTest` to something that not conflicts with the kotlin test `runTest` method? That's part of the confusion.

To enforce the required ordering, you can use a [Semaphore](https://kotlinlang.org/api/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines.sync/-semaphore/)

Comment (mhuster23) [reply model-server/src/test/kotlin/org/modelix/model/server/V1ApiTest.kt]: done

**Review by languitar — APPROVED** [review]
Ah, good idea with the plugin.

The only thing potentially left is that we have the exact same logic now copied to two tests, which could be deduplicated.

**Review by languitar — APPROVED** [review]
